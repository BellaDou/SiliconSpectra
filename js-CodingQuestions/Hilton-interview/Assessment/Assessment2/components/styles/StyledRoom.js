import styled from "styled-components";

const StyledRoom = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 140px;
  border-style: solid;
  ${({ room }) => {
    if (!room.ableToCheck || room.checked) {
      return `
      border-color: #ededed;
    `;
    } else {
      return `border-color: #b8b8b8;`;
    }
  }}
  border-radius: 5px;
  margin-bottom: 10px;
  margin-right: 10px;
`;

export default StyledRoom;
