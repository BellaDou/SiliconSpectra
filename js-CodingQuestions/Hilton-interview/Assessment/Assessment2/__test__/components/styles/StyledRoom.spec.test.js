import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import StyledRoom from "../../../components/styles/StyledRoom";

const room = {
  id: 2,
  ableToCheck: true,
  checked: true,
  adults: 1,
  children: 0
};

test("Should properly render style component", () => {
  const tree = renderer.create(<StyledRoom room={room} />).toJSON();
  expect(tree).toMatchSnapshot();
});
