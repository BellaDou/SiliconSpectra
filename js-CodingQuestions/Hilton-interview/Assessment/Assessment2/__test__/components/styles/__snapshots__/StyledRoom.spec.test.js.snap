// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should properly render style component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 140px;
  border-style: solid;
  border-color: #ededed;
  border-radius: 5px;
  margin-bottom: 10px;
  margin-right: 10px;
}

<div
  className="c0"
/>
`;
